scipy/interpolate/_fitpack2.py

```diff
@@ -1917,8 +1917,6 @@ def __init__(self, theta, phi, r, w=None, s=0., eps=1E-16):
         if not 0.0 < eps < 1.0:
             raise ValueError('eps should be between (0, 1)')
 
-        if np.issubclass_(w, float):
-            w = ones(len(theta)) * w
```

Comment:
@rgommers I claim that this `if` branch should be removed. 

`w` parameter is defined as `array_like, optional`. If it isn't None, then it's being set to `w = np.asarray(w)`. As a result `w` can end up as `None` or a numpy arrray.

Therefore, `np.issubclass_(w, float)` will always yield `False`, as only classes are accepted as arguments.  

Even hypothetically, if execution would have entered inside if statement, then:
1. `w` would have to be a class (`type(w) is type`)
2. Operation inside if statement would always fail: `np.ones(3) * float` raises `TypeError`.

Reply:
That looks right, good catch.